### PR TITLE
[config] improve root-config --has handling

### PR DESCRIPTION
**Pull request by SpectreVert**
Addressing issue #7662 for bash root-config

Pretty much mirrored what was done for root-config.bat in #7663

Handle unknown feature. Unknown option is already handled.

Preview:

spectre@localhost ~/root/compiled/bin
 (improve-root-config-bash)> ./root-config --has-xml --has-cxx14 --has-improved
yes yes --has-improved: unknown feature!

### CODE REVIEW

**Comment by bellenot** [comment]
@SpectreVert please fork also [roottest](https://github.com/root-project/roottest) to prevent timeouts on the Windows nodes. Thanks

File: config/root-config.in [hunk 1]
```diff
@@ -469,16 +470,26 @@ while test $# -gt 0; do
       ;;
     --has-*)
       ### Check for feature
-      f=`echo $1 | sed 's/--has-//'`
-      for c in $features ; do
-         if test "x$c" = "x$f" ; then
+      target=`echo $1 | sed 's/--has-//'`
```
Comment (Axel-Naumann) [inline config/root-config.in:473]: Can you try what happens with `root-config --has-`
Is there a syntax error, do we say "yes", or do we handle that in a sane way?

File: config/root-config.in [hunk 2]
```diff
@@ -469,16 +470,26 @@ while test $# -gt 0; do
...
          fi
-         c=""
+         feature=""
       done
-      if test "x$c" = "x" ; then
-         out="$out no"
+      if test "$feature" = "" ; then
+			for feature in $all_features ; do
+				if test "$feature" = "$target" ; then
+					found="$feature"
```
Comment (Axel-Naumann) [inline config/root-config.in:485]:
```suggestion
					found="$feature"
					break
```

File: config/root-config.in [hunk 3]
```diff
@@ -469,16 +470,26 @@ while test $# -gt 0; do
...
+				fi
+			done
+			if test "$found" = "" ; then
+				out="$out --has-$target: unknown feature!"
```
Comment (Axel-Naumann) [inline config/root-config.in:489]: Shall this switch the exit code to `1`, or maybe even `exit 1`?

File: config/root-config.in [hunk 4]
```diff
@@ -48,6 +48,7 @@ libdir=@libdir@
 incdir=@incdir@
 etcdir=@etcdir@
 tutdir=@tutdir@
+all_features="@all_features"
```
Comment (Axel-Naumann) [inline config/root-config.in:51]:
```suggestion
all_features="@all_features@"
```
? Please make sure this actually works before submitting the PR :-) (Maybe you did, just stating the "obvious")

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Wonderful thank you! Couple of small suggestions...

File: config/root-config.in [hunk 5]
```diff
@@ -469,16 +470,32 @@ while test $# -gt 0; do
...
-         out="$out no"
+      if test "$feature" = "" ; then
+         for feature in $all_features ; do
+            if test "$feature" = "$target" ; then
+               found="$feature"
+               break
+            fi
+         done
+         if test "$found" = "" ; then
+            errors="$errors--has-$target: unknown feature!\n"
```
Comment (Axel-Naumann) [inline config/root-config.in:495]: Nice solution! But maybe, just to be on the safe side:
```suggestion
            errors="$errors --has-$target: unknown feature!\n"
```
or
```suggestion
            errors="${errors}--has-$target: unknown feature!\n"
```

**Review by Axel-Naumann — APPROVED** [review]
Wow, super nice! :heart:ing it!

**Comment by Axel-Naumann** [comment]
Let's wait for Jenkins to finish!

**Comment by SpectreVert** [comment]
If we want to be thourough there's another line where interpolation could fail. Let me fix it and re-commit.

**Comment by SpectreVert** [comment]
Done. I've commited my last words

**Pull request merged**